extend/component/implementation/python.md

```diff
@@ -111,82 +157,254 @@ with open('in/tables/source.csv', mode='rt', encoding='utf-8') as in_file, open(
+This is especially useful because the KBC input mapping will by default include all versions of files matching specific tag. By default, the method 
+returns only the latest file of each.
+
+```python
+from keboola.component import CommonInterface
+import logging
+
+# init the interface
+ci = CommonInterface()
+
+input_files = ci.get_input_files_definitions(tags= ['my_tag'], only_latest_files=True)
+
+# print path of the first file (random order) matching the criteria
+first_file = input_files[0]
+logging.info(f'The first file named: "{input_files.name}" is at path: {input_files.full_path}')
+```
+
+
+When working with files it may be useful to retrieve them in a dictionary structure grouped either by name or a tag group. 
+For this there are convenience methods `get_input_file_definitions_grouped_by_tag_group()` and `get_input_file_definitions_grouped_by_name()`
+
+
+```python
+from keboola.component import CommonInterface
+import logging
+
+# init the interface
+ci = CommonInterface()
+
+# group by tag
+input_files_by_tag = ci.get_input_file_definitions_grouped_by_tag_group(only_latest_files=True)
+
+# print list of files matching specific tag
+logging.info(input_files_by_tag['my_tag']) 
+
+# group by name
+input_files_by_name = ci.get_input_file_definitions_grouped_by_name(only_latest_files=True)
+
+# print list of files matching specific name
+logging.info(input_files_by_name['image.jpg'])
+
+```
+
+#### Processing state files
+
+[State files](/extend/common-interface/config-file/#state-file) can be easily loaded and written 
+using the `get_state_file()` and `write_state_file()` methods:
+ 
+```python
+from keboola.component import CommonInterface
+from datetime import datetime
+import logging
+
+# init the interface
+ci = CommonInterface()
+
+last_state = ci.get_state_file()
+
+# print last_updated if exists
+logging.info(f'Previous job stored following last_updated value: {last_state.get("last_updated","")})')
+
+# store new state file
+ci.write_state_file({"last_updated": datetime.now().isoformat()})
+```
+
+### Logging
+
+The library automatically initializes STDOUT or GELF logger based on the presence of the `KBC_LOGGER_PORT/HOST` environment variables 
+upon the `CommonInterface` initialization. To use the GELF logger just enable the logger for your application in the 
+[Developer Portal](https://components.keboola.com/). 
+More details about logging options are available in a [dedicated article](/extend/common-interface/logging/#examples).
+
+With either setting, you can log your messages using the logging library:
+
+```python
+from keboola.component import CommonInterface
+from datetime import datetime
+import logging
+
+# init the interface
+ci = CommonInterface()
+
+logging.info("Info message")
+```
+
+You may also choose to override the settings by enabling the GELF or STDOUT explicitly and specifying the host/port parameters:
```

Comment:
I would completely delete this, why would you want to do that? It looks only like a good source of errors :)

Reply:
Agreed.

Reply:
Removed in 9cd3e60